commands2/subsystem.py

```diff
@@ -28,15 +28,16 @@ class Subsystem(Sendable):
     base for user implementations that handles this.
     """
 
-    def __new__(cls, *arg, **kwargs) -> "Subsystem":
-        instance = super().__new__(cls)
-        super().__init__(instance)
-        SendableRegistry.addLW(instance, cls.__name__, cls.__name__)
-        # add to the scheduler
-        from .commandscheduler import CommandScheduler
-
-        CommandScheduler.getInstance().registerSubsystem(instance)
-        return instance
+    if not TYPE_CHECKING:
+        def __new__(cls, *arg, **kwargs) -> "Subsystem":
```

Comment:
This seems wrong. We should return `typing.Self` instead.

Reply:
I agree. But typing.Self was added in 3.11 according to https://docs.python.org/3/library/typing.html#typing.Self

Reply:
We're already using typing_extensions in command, so let's just use it here?

Reply:
Ok, swapped it over